gaurav/src/components/MarkTime.tsx

```diff
@@ -45,35 +51,53 @@ export const MarkTime = () => {
     const time = new Date();
     const punchOutTime = time.getTime();
     const punchInTime = new Date(punchData[punchData.length - 1].time).getTime();
-    console.log("success");
-    if (punchData.length === 0 || punchData[punchData.length - 1].type === "IN") {
-      const newPunchOutData = [
-        ...punchData,
-        {
-          id: punchData.length,
-          time: new Date().toISOString(),
-          type: "OUT",
-        },
-      ];
-      setPunchData(newPunchOutData);
-      localStorage.setItem("punchData", JSON.stringify(newPunchOutData));
-
-      const newNetTime = (punchOutTime - punchInTime) / 1000 + netTime;
-      setNetTime(newNetTime); // time in seconds
-      localStorage.setItem("netTime", JSON.stringify(newNetTime));
+    if ((punchOutTime - punchInTime) / 1000 < 120) {
+      setIsError(true);
+    } else {
+      console.log("success");
+      if (punchData.length === 0 || punchData[punchData.length - 1].type === "IN") {
+        const newPunchOutData = [
+          ...punchData,
```

Comment:
reuse already written logic using function modularity instead of copy and pasting... code duplication is bad

Reply:
created `lastPunchData = punchData[punchData.length - 1]`
and used it int the place of  `punchData[punchData.length - 1].type === "IN"` i.e `lastPunchData.type === "IN`